test/spec/test_case.zig

```diff
@@ -0,0 +1,106 @@
+const std = @import("std");
+const snappy = @import("snappy");
+const ForkSeq = @import("params").ForkSeq;
+const isFixedType = @import("ssz").isFixedType;
+const BeaconStateAllForks = @import("state_transition").BeaconStateAllForks;
+
+const consensus_types = @import("consensus_types");
+const phase0 = consensus_types.phase0;
+const altair = consensus_types.altair;
+const bellatrix = consensus_types.bellatrix;
+const capella = consensus_types.capella;
+const deneb = consensus_types.deneb;
+const electra = consensus_types.electra;
+
+// TODO: Most of the runners have fixed ssz types ie. Given a dir, we can already derive which
+// snappy file has what type and have this info written in test_type/schema.
+// There is an exception which is `transition`. One needs to read meta.yaml to figure out
+// whether a block_x.ssz_snappy belong to old fork or new fork. Current design of loadTestCase
+// does not handle this.
+pub fn loadTestCase(comptime Schema: type, comptime SchemaOut: type, dir: std.fs.Dir, allocator: std.mem.Allocator) !SchemaOut {
+    var out: SchemaOut = .{};
```

Comment:
there is some copy overhead here
having `out` params will resolve that and go with the convention in [ssz-z](https://github.com/ChainSafe/ssz-z/blob/76cde3c7ea23b33d2789e8e372003272abe0e335/src/ssz/type/container.zig#L82)
also we want the consumer to take ownership here